### PR TITLE
Time of the day - correction (issue/8247)

### PR DESCRIPTION
**Pull request by kingfisher88**
Remove "anonymous (not verified)" from greeting message for anonymous users.
Change the service that returns the current time of day, returns just the
greeting message according to the time of day.

### CODE REVIEW

File: web/modules/custom/usergreeting/src/GreetingTime.php [hunk 1]
```diff
@@ -31,17 +31,10 @@ final class GreetingTime {
...
-   */
-  protected $account;
-
   /**
    * The string to translation.
    *
-   * @var \Drupal\usergreeting\Drupal\Core\StringTranslation\TranslationInterface
+   * @var \Drupal\Core\StringTranslation\TranslationInterface
    */
   protected $stringTranslation;
```
Comment (lekob) [inline web/modules/custom/usergreeting/src/GreetingTime.php]: The $stringTranslation property is defined in StringTranslationTrait. No need to redefine it here.

File: web/modules/custom/usergreeting/src/GreetingTime.php [hunk 2]
```diff
@@ -70,7 +60,7 @@ public function __construct(ConfigFactoryInterface $config_factory, TimeInterfac
    * @return string
```
Comment (lekob) [inline web/modules/custom/usergreeting/src/GreetingTime.php]: You are not returning a string.

File: web/modules/custom/usergreeting/src/GreetingTime.php [hunk 3]
```diff
@@ -70,7 +60,7 @@ public function __construct(ConfigFactoryInterface $config_factory, TimeInterfac
...
    *   The greeting message output.
    */
-  public function greetingMessage(): array {
+  public function greetingMessage(): TranslatableMarkup {
     $time_output = (int) date('G', $this->timeOutput->getRequestTime());
     $config_morning = $this->configFactory->get('usergreeting.settings')->get('morning_start');
     $config_afternoon = $this->configFactory->get('usergreeting.settings')->get('afernoon_start');
```
Comment (lekob) [inline web/modules/custom/usergreeting/src/GreetingTime.php]: afternoon

File: web/modules/custom/usergreeting/src/Plugin/Block/UsergreetingBlock.php [hunk 4]
```diff
@@ -38,33 +46,52 @@ class UsergreetingBlock extends BlockBase implements ContainerFactoryPluginInter
...
-      $container->get('usergreeting.greeting')
+      $container->get('usergreeting.greeting'),
+      $container->get('current_user')
     );
   }
 
   /**
    * {@inheritdoc}
    */
   public function build() {
```
Comment (lekob) [inline web/modules/custom/usergreeting/src/Plugin/Block/UsergreetingBlock.php]: Declare array return type.

**Review by lekob — CHANGES_REQUESTED** [review]

File: web/modules/custom/usergreeting/src/GreetingTime.php (hunk 1 above)
Comment (gkurucz) [reply web/modules/custom/usergreeting/src/GreetingTime.php:39]: According to https://api.drupal.org/api/drupal/core!lib!Drupal!Core!StringTranslation!StringTranslationTrait.php/trait/StringTranslationTrait/8.8.x he should.

> If the class is capable of injecting services from the container, it should inject the 'string_translation' service and assign it to $this->stringTranslation.

I've also seen this exact same practice in core.

File: web/modules/custom/usergreeting/src/Plugin/Block/UsergreetingBlock.php [hunk 5]
```diff
@@ -38,33 +46,52 @@ class UsergreetingBlock extends BlockBase implements ContainerFactoryPluginInter
...
-    $build = [];
+  public function build(): array {
     $greetuser = $this->greeting->greetingMessage();
 
-    $build['greetuser'] = $greetuser;
-
-    $build['#cache']['max-age'] = 0;
-    return $build;
-
+    if ($this->account->isAnonymous() === TRUE) {
```
Comment (gkurucz) [inline web/modules/custom/usergreeting/src/Plugin/Block/UsergreetingBlock.php]: This condition can be simplified.

```
$message = [
  common_stuff...
];

if (condition) {
  $message['different_stuff'] = 'different_stuff';
  $message['stuff_to_change'] = 'other_different_stuff';
}
```

**Review by gkurucz — APPROVED** [review]

**Review by lekob — APPROVED** [review]

**Pull request merged**